### PR TITLE
refactor: extract shared primitives into vite_powershell crate

### PR DESCRIPTION
**Pull request by fengmk2**
## Summary

- New `vite_powershell` crate owns the platform-shared primitives previously inlined in `vite_task_plan::ps1_shim`: `POWERSHELL_PREFIX`, `powershell_host()` (LazyLock-cached `pwsh.exe`/`powershell.exe` lookup), and `find_ps1_sibling()` (case-insensitive `.cmd` → sibling `.ps1` resolution).
- `vite_task_plan::ps1_shim` keeps its task-specific layers — workspace + `node_modules/.bin` scope check, cwd-relative arg conversion, `Arc<AbsolutePath>` / `Arc<[Str]>` return types — and now imports the primitives from `vite_powershell`.
- No behavior change. All existing snapshot/unit/integration tests pass.

## Why

vite-plus is adding `vite_command::ps1_shim` (see voidzero-dev/vite-plus#1489) to fix the same Ctrl+C terminal-corruption symptom for package-manager-routed commands (`vp dlx`, `vp add`, `vp update`, etc.). Without this extraction it would have to copy the constant, the host-discovery LazyLock, and the sibling-`.ps1` check. With this extraction both task scripts (vite-task) and pm commands (vp) compose the same primitives with their own scope rules.

The two callers diverge intentionally and don't share enough to merge into one helper:
- vite-task's wrapper produces `(Arc<AbsolutePath>, Arc<[Str]>)` with **cwd-relative** `.ps1` paths because the args become part of the task graph's spawn fingerprint and must stay portable across machines.
- vite-plus's wrapper (in vp's `vite_command`) produces `(AbsolutePathBuf, Vec<OsString>)` with **absolute** `.ps1` paths because there's no cache fingerprint at spawn time.

## Test plan

- [x] `cargo test -p vite_powershell` — 5 new tests pass (sibling found, case-insensitive `.CMD`, no-sibling, non-`.cmd`, no-extension)
- [x] `cargo test -p vite_task_plan` — 39 unit + 45 integration tests pass; the existing rewrite snapshot tests are preserved
- [x] `cargo clippy -p vite_powershell -p vite_task_plan --all-targets` — clean
- [x] Companion vp PR to switch `vite_command::ps1_shim` to depend on `vite_powershell` once this lands and a new vite-task SHA is pinned

---

> [!NOTE]
> **Low Risk**
> Mostly a dependency extraction/refactor with tests; behavior should remain the same aside from potential edge-case differences in `.cmd`/`.ps1` detection on Windows.
> 
> **Overview**
> Extracts the Windows `.cmd`→`.ps1` PowerShell-shim primitives into a new `vite_powershell` crate, including the `POWERSHELL_PREFIX` args, cached `pwsh.exe`/`powershell.exe` host discovery, and sibling-`.ps1` lookup.
> 
> Updates `vite_task_plan::ps1_shim` to depend on and call these shared helpers while keeping its workspace/`node_modules/.bin` scoping and cwd-relative argument rewrite logic, and adds unit tests for the new crate. Workspace `Cargo.toml`/`Cargo.lock` are updated to include the new crate and dependency.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1e2b7913840c15aa670d702f8bf4c7fdcd45c56f. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: crates/vite_task_plan/src/ps1_shim.rs [hunk 1]
```diff
@@ -21,24 +21,21 @@
...
 //! See <https://github.com/voidzero-dev/vite-plus/issues/1176>.
 
 use std::sync::Arc;
 
 #[cfg(any(windows, test))]
 use cow_utils::CowUtils as _;
 use vite_path::AbsolutePath;
-#[cfg(any(windows, test))]
+#[cfg(test)]
 use vite_path::AbsolutePathBuf;
```
Comment (fengmk2) [inline crates/vite_task_plan/src/ps1_shim.rs]: move this line to `#[cfg(test)]` block inside.

**Comment by fengmk2** [comment]
@cursor review

**Review by wan9chi — APPROVED** [review]
